lib/arel_extensions/visitors/mssql.rb

```diff
@@ -2,52 +2,68 @@ module ArelExtensions
   module Visitors
     module MSSQL
 
-      mssql_class = Arel::Visitors.constants.select { |c|
-        Arel::Visitors.const_get(c).is_a?(Class) && %i[MSSQL SQLServer].include?(c)
-      }.first
-
-      LOADED_VISITOR = Arel::Visitors.const_get(mssql_class) || Arel::Visitors.const_get('MSSQL')
-
-      LOADED_VISITOR::DATE_MAPPING = {
-        'd' => 'day', 'm' => 'month', 'y' => 'year', 'wd' => 'weekday', 'w' => 'week', 'h' => 'hour', 'mn' => 'minute', 's' => 'second'
-      }.freeze
-
-      LOADED_VISITOR::DATE_FORMAT_DIRECTIVES = {
-        '%Y' => 'YYYY', '%C' => '', '%y' => 'YY', '%m' => 'MM', '%B' => 'month', '%^B' => '', '%b' => '', '%^b' => '', # year, month
-        '%V' => 'iso_week', '%G' => '',                                                                                # ISO week number and year of week
-        '%d' => 'DD', '%e' => ''  , '%j' => ''  , '%w' => 'dw', %'a' => '', '%A' => 'weekday',                         # day, weekday
-        '%H' => 'hh', '%k' => ''  , '%I' => ''  , '%l' => ''  , '%P' => '', '%p' => '',                                # hours
-        '%M' => 'mi', '%S' => 'ss', '%L' => 'ms', '%N' => 'ns', '%z' => 'tz'
-      }.freeze
-
-      LOADED_VISITOR::DATE_FORMAT_FORMAT = {
-        'YY' => '0#', 'MM' => '0#', 'DD' => '0#', 'hh' => '0#', 'mi' => '0#', 'ss' => '0#', 'iso_week' => '0#'
-      }
-
-      LOADED_VISITOR::DATE_NAME = [
-        '%B', '%A'
-      ]
-
-      LOADED_VISITOR::DATE_FORMAT_REGEX =
-        Regexp.new(
-          LOADED_VISITOR::DATE_FORMAT_DIRECTIVES
-            .keys
-            .map{|k| Regexp.escape(k)}
-            .join('|')
-        ).freeze
-
-      # TODO; all others... http://www.sql-server-helper.com/tips/date-formats.aspx
-      LOADED_VISITOR::DATE_CONVERT_FORMATS = {
-        'YYYY-MM-DD' => 120,
-        'YY-MM-DD' => 120,
-        'MM/DD/YYYY' => 101,
-        'MM-DD-YYYY' => 110,
-        'YYYY/MM/DD' => 111,
-        'DD-MM-YYYY' => 105,
-        'DD-MM-YY'   => 5,
-        'DD.MM.YYYY' => 104,
-        'YYYY-MM-DDTHH:MM:SS:MMM' => 126
-      }.freeze
+      MSSQL_CLASS_NAMES = %i[MSSQL SQLServer].freeze
+
+      mssql_class =
```

Comment:
Why not a constant ?

Reply:
Does it change a lot in this case? We're only going to use it for const_get later.

Reply:
If we don't intend to expose it in descendants, you are right it should stay a local var.